src/consensus/pbft_manager.cpp

```diff
@@ -1455,20 +1455,29 @@ std::pair<vec_blk_t, bool> PbftManager::comparePbftBlockScheduleWithDAGblocks_(P
 
 bool PbftManager::pushCertVotedPbftBlockIntoChain_(taraxa::blk_hash_t const &cert_voted_block_hash,
                                                    std::vector<Vote> const &cert_votes_for_round) {
-  if (!checkPbftBlockValid_(cert_voted_block_hash)) {
-    syncPbftChainFromPeers_(invalid_cert_voted_block, cert_voted_block_hash);
-    return false;
-  }
   auto pbft_block = pbft_chain_->getUnverifiedPbftBlock(cert_voted_block_hash);
   if (!pbft_block) {
-    LOG(log_er_) << "Can not find the cert vote block hash " << cert_voted_block_hash << " in pbft queue";
+    pbft_block = db_->getPbftCertVotedBlock(cert_voted_block_hash);
+    if (!pbft_block) {
+      LOG(log_nf_) << "Can not find the cert voted block hash " << cert_voted_block_hash
+                   << " in both pbft queue and DB";
+      return false;
+    }
+    // Read from DB pushing into unverified queue
+    pbft_chain_->pushUnverifiedPbftBlock(pbft_block);
```

Comment:
If we are pushing block back to the unverified queue only in scenarios when the node got restarted so it does not have it memory, please comment it properly because I doubt that anyone would know why is this here except you...

Or put some comment above `if (!pbft_block)` condition if everything inside is related to the node restart scenario

Reply:
yeah, add comments